application/common/models/User.php

```diff
@@ -319,33 +320,78 @@ public function checkAndProcessHIBP(): void
             return;
         }
 
-        try {
-            if (!HIBP::isPwned($this->password)) {
-                $this->currentPassword->extendHibpCheckAfter();
-                return;
-            }
-        } catch (Exception $e) {
-            \Yii::error([
-                'action' => 'check and process hibp',
-                'employee_id' => $this->employee_id,
-                'message' => $e->getMessage(),
-            ]);
+        if (! $this->isPasswordPwned()) {
+            $this->currentPassword->extendHibpCheckAfter();
+            return;
+        }
 
+        if (\Yii::$app->params['hibpTrackingOnly']) {
+            // extend check after date to only track user once per checking period
+            $this->currentPassword->extendHibpCheckAfter();
+            $this->trackPwnedPasswordGAEvent();
```

Comment:
Can the `trackPwnedPasswordGAEvent` call be moved above the `if (\Yii::$app->params['hibpTrackingOnly'])` and line 336 be removed, since it's called regardless of the tracking-only param?

Reply:
sure, will do.